android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java

```diff
@@ -38,9 +50,174 @@ public void justSomeInteraction() {
         onView(withId(R.id.tabLayout)).check(matches(isDisplayed()));
         onView(withId(R.id.store_text_layout)).check(matches(isDisplayed()));
         onView(withId(R.id.store_text)).check(matches(isDisplayed()));
-        onView(withId(R.id.store_text)).perform(typeText("123"), ViewActions.closeSoftKeyboard());
+        onView(withId(R.id.store_text)).perform(typeText("123"), closeSoftKeyboard());
         onView(withId(R.id.category_text)).perform(click());
     }
 
+    @Test
+    public void testInvalidInput() {
```

Comment:
Here it would be useful to split the test a bit. There are many scenarios testet in one single test. 
e.g. `testInvalidIncomeInputs()`, `testInvalidExpenseInputs()`, `testMultipleInvalidInputs()`.
Obviously, testing for every combination of inputs seems to be a bit much right now, but it would be helpful if there are more smaller tests instead of one big test